### PR TITLE
Ensure annotations on empty entries in notebook are not lost

### LINKED ISSUE #6156: Annotations on empty entries in notebook are lost
#### Summary
Entries do not get persisted until text is entered. If you create a blank entry, then add an annotation/tag to it, then navigate away, the blank entry is removed and there is a misleading indication that the tag is attached to something.

#### Expected vs Current Behavior
Either the tag is also removed from view, or blank entries can be persisted. @charlesh88 , what say you?

#### Steps to Reproduce
1. create and navigate to a notebook
2. click + to add an entry
3. click on the entry element
4. click on the annotations tab in inspector
5. add an annotation/tab
6. observe the blank entry is removed from view and the tag in inspector is still displayed

#### Environment

* Open MCT Version: master
* Deployment Type: 
* OS:
* Browser:

#### Impact Check List

- [x] Data loss or misrepresented data?
- [] Regression? Did this used to work or has it always been broken?
- [] Is there a workaround available?
- [] Does this impact a critical component?
- [x] Is this just a visual bug with no functional impact?
- [] Does this block the execution of e2e tests?
- [] Does this have an impact on Performance?

### PR DESCRIPTION
**Pull request by scottbell**
Closes #6156 #6411 

### Describe your changes:
Don't (sometimes) delete entries if text is empty. Select and focus entries when created.

### All Submissions:

* [x] Have you followed the guidelines in our [Contributing document](https://github.com/nasa/openmct/blob/master/CONTRIBUTING.md)?
* [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/nasa/openmct/pulls) for the same update/change?
* [x] Is this change backwards compatible? For example, developers won't need to change how they are calling the API or how they've extended core plugins such as Tables or Plots.

### Author Checklist

* [x] Changes address original issue?
* [x] Tests included and/or updated with changes?
* [x] Command line build passes?
* [x] Has this been smoke tested?
* [x] Testing instructions included in associated issue OR is this a dependency/testcase change?

### Reviewer Checklist

* [x] Changes appear to address issue?
* [x] Reviewer has tested changes by following the provided instructions?
* [x] Changes appear not to be breaking changes?
* [x] Appropriate automated tests included?
* [x] Code style and in-line documentation are appropriate?
* [x] Has associated issue been labelled unverified? (only applicable if this PR closes the issue)
* [x] Has associated issue been labelled bug? (only applicable if this PR is for a bug fix)

### CODE REVIEW

**Review by unlikelyzero — CHANGES_REQUESTED** [review]
This looks great, but there are several references in this script to remove 
https://github.com/nasa/openmct/blob/e57eedc3c8cdf3d7cbd746646e5f657eca5ec9c1/e2e/tests/functional/plugins/notebook/tags.e2e.spec.js#L134-L135

File: e2e/tests/functional/plugins/notebook/tags.e2e.spec.js [hunk 1]
```diff
@@ -110,6 +110,24 @@ test.describe('Tagging in Notebooks @addInit', () => {
         await expect(page.locator('[aria-label="Autocomplete Options"]')).not.toContainText("Driving");
         await expect(page.locator('[aria-label="Autocomplete Options"]')).toContainText("Drilling");
     });
+    test('Can add tags with blank entry', async ({ page }) => {
+        //Go to baseURL
+        await page.goto('./', { waitUntil: 'networkidle' });
```
Comment (unlikelyzero) [inline e2e/tests/functional/plugins/notebook/tags.e2e.spec.js:115]: usually this is common to all beforeEach steps

File: e2e/tests/functional/plugins/notebook/notebook.e2e.spec.js [hunk 2]
```diff
@@ -263,7 +271,25 @@ test.describe('Notebook entry tests', () => {
...
+    test('previous and new entries can be deleted', async ({ page }) => {
+        // Navigate to the notebook object
+        await page.goto(notebookObject.url);
+
+        await nbUtils.enterTextEntry(page, 'First Entry');
+        await page.hover('text="First Entry"');
+        await page.click('button[title="Delete this entry"]');
+        await page.getByRole('button', { name: 'Ok' }).filter({ hasText: 'Ok' }).click();
+        await expect(page.locator('text="First Entry"')).toBeHidden();
+        await nbUtils.enterTextEntry(page, 'Another First Entry');
```
Comment (unlikelyzero) [inline e2e/tests/functional/plugins/notebook/notebook.e2e.spec.js:283]: whoa this is awesome

File: src/plugins/notebook/components/Notebook.vue [hunk 3]
```diff
@@ -793,15 +793,29 @@ export default {
...
+            const entryAnnotations = this.notebookAnnotations[id] ?? {};
+            selectEntry({
+                element,
+                entryId: id,
+                domainObject: this.domainObject,
+                openmct: this.openmct,
+                notebookAnnotations: entryAnnotations
+            });
+            if (event) {
+                event.stopPropagation();
```
Comment (unlikelyzero) [inline src/plugins/notebook/components/Notebook.vue:813]: how would we cover this line?

File: e2e/tests/functional/plugins/notebook/tags.e2e.spec.js (hunk 1 above)
Comment (scottbell) [reply e2e/tests/functional/plugins/notebook/tags.e2e.spec.js:115]: It's true, but the tests above are also adding notebook entries with text, which we don't want for this particular test.

File: src/plugins/notebook/components/Notebook.vue (hunk 3 above)
Comment (scottbell) [reply src/plugins/notebook/components/Notebook.vue:813]: I'm surprised this line isn't being covered as it was caught by tests! Ask @ozyx 😀

File: e2e/tests/functional/plugins/notebook/tags.e2e.spec.js [hunk 4]
```diff
@@ -81,10 +78,13 @@ async function createNotebookEntryAndTags(page, iterations = 1) {
...
 test.describe('Tagging in Notebooks @addInit', () => {
+    test.beforeEach(async ({ page }) => {
+        //Go to baseURL
+        await page.goto('./', { waitUntil: 'networkidle' });
+    });
     test('Can load tags', async ({ page }) => {
         await createNotebookAndEntry(page);
 
-        // TODO can be removed with fix for https://github.com/nasa/openmct/issues/6411
         await page.locator('[aria-label="Notebook Entry"].is-selected div.c-ne__text').click();
```
Comment (unlikelyzero) [inline e2e/tests/functional/plugins/notebook/tags.e2e.spec.js:88]: remove these

**Review by unlikelyzero — APPROVED** [review]

**Review by ozyx — APPROVED** [review]
LGTM

**Pull request merged**